### PR TITLE
Add get_download_url to abstract urls for firmware download

### LINKED ISSUE #303: Navigator Ardusub binary is not being chmodded +x ?
I removed the binary to trigger a re-download.
This was the result in the autopilot session:
image

it worked after I manually chmodded the file

### PR DESCRIPTION
**Pull request by patrickelectric**
Fix #303
Alternative to #304 

Adds a function to find the correct URL to download, this allow us to have magic urls for specific products if a desired combination is valid. Also unifies the validation and permission between such options.

### CODE REVIEW

File: core/services/ardupilot_manager/firmware_download/FirmwareDownload.py [hunk 1]
```diff
@@ -229,10 +228,10 @@ def download(self, vehicle: Vehicle, platform: Platform, version: str = "") -> O
                 Defaults to None.
 
         Returns:
-            Optional[pathlib.Path]: Temporary path for the firmware file, None if unable to download or validate file.
+            Optional[pathlib.Path]: URL of valid firmware or None if there are not such thing.
```
Comment (Williangalvani) [inline core/services/ardupilot_manager/firmware_download/FirmwareDownload.py:231]: Optional[str]

**Review by Williangalvani — CHANGES_REQUESTED** [review]

File: core/services/ardupilot_manager/firmware_download/FirmwareDownload.py (hunk 1 above)
Comment (rafaellehmkuhl) [reply core/services/ardupilot_manager/firmware_download/FirmwareDownload.py]: "if there is no such thing"

Comment (patrickelectric) [reply core/services/ardupilot_manager/firmware_download/FirmwareDownload.py]: Done.

**Review by rafaellehmkuhl — CHANGES_REQUESTED** [review]

File: core/services/ardupilot_manager/firmware_download/FirmwareDownload.py [hunk 2]
```diff
@@ -229,10 +228,10 @@ def download(self, vehicle: Vehicle, platform: Platform, version: str = "") -> O
                 Defaults to None.
 
         Returns:
-            Optional[pathlib.Path]: Temporary path for the firmware file, None if unable to download or validate file.
+            Optional[str]: URL of valid firmware or None if there are no such thing.
```
Comment (rafaellehmkuhl) [inline core/services/ardupilot_manager/firmware_download/FirmwareDownload.py:231]: "there is no such thing."

**Review by rafaellehmkuhl — CHANGES_REQUESTED** [review]

**Review by rafaellehmkuhl — APPROVED** [review]

**Review by Williangalvani — APPROVED** [review]

**Pull request merged**